src/hooks/AddressSummary.ts

```diff
@@ -0,0 +1,31 @@
+import { useMemo } from 'react'
+import { WalletInfo, AddressStatus } from '../context/WalletContext'
+import { BitcoinAddress } from '../libs/JmWalletApi'
+
+type AddressInfo = {
+  status: AddressStatus
+  address: BitcoinAddress
+}
+
+type AddressSummary = {
+  [key: BitcoinAddress]: AddressInfo
+}
+
+const useAddressSummary = (currentWalletInfo: WalletInfo | null): AddressSummary | null => {
```

Comment:
```suggestion
const useAddressSummary = (currentWalletInfo: WalletInfo | null): AddressSummary | null => {
```

What's the reasoning behind allowing `currentWalletInfo` to be `null`? Everywhere where we use `useWalletSummary` we do a non-null assertion (`!`) anyway. Would it work to just do this signature and remove these assertions?

``` suggestion
const useAddressSummary = (currentWalletInfo: WalletInfo): AddressSummary => {
```

Reply:
Yes, you are right. This would be possible. I was thinking, as this is currently implemented as a hook, following scenario can occur:
```
const currentWallet = useCurrentWallet()
const addressSummary = useAddressSummary(walletInfo)
```
Since `useCurrentWallet` can return `null`, `useAddressSummary` should also accept it as parameter..

But you are completely right, this should never happen in this case, as it is even prevented to load some pages (Send, Earn, etc.) if no wallet is present.

The upcoming changes in https://github.com/joinmarket-webui/joinmarket-webui/pull/383 already do away the `null` param. Would it be okay for you, if you are satisfied with the changes in general, to close this PR in favor of #383?

Reply:
Would you prefer to have #383 replace this PR?

Reply:
Yes.